### PR TITLE
Search complex feature And filter

### PR DESCRIPTION
**Pull request by ikechiafrics**
Implemented a search endpoint from the API, now the user can search for any food which is on the Spoonacular Api. Chnged the edit text search to a search view, and Added a filter feature, which filters out the food item searched.
This filters the searched based on the intolerances such as gluten, peanut, seafood etc. Sorts based on popularity, price, and cholesterol, sort direction(Ascending or Descending), and the number of items which are to be displayed.

https://user-images.githubusercontent.com/93938295/180101802-9113c481-d178-4292-bf26-d1215171984b.mov

### CODE REVIEW

**Review by lineilfb — APPROVED** [review]

**Review by lineilfb — APPROVED** [review]

File: app/src/main/java/com/example/metachef/Adapters/ComplexSearchAdapter.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View itemsView = LayoutInflater.from(context).inflate(R.layout.item_complexsearch, parent, false);
+        return new ComplexSearchAdapter.ViewHolder(itemsView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        int roundingRadius = 50;
```
Comment (JamesMcGaa) [inline app/src/main/java/com/example/metachef/Adapters/ComplexSearchAdapter.java:44]: magic number

File: app/src/main/java/com/example/metachef/Fragments/SearchFragment.java [hunk 2]
```diff
@@ -78,4 +109,184 @@ public void diderror(String message) {
...
+                    Intolerances.remove("Wheat");
+                }
+            }
+        });
+        checkSeafood = view.requireViewById(R.id.checkSeafood);
+        checkSeafood.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if(checkSeafood.isChecked()){
+                    Intolerances.add("Seafood");
```
Comment (JamesMcGaa) [inline app/src/main/java/com/example/metachef/Fragments/SearchFragment.java:204]: should be an enum or const

File: app/src/main/java/com/example/metachef/RequestManager.java [hunk 3]
```diff
@@ -47,9 +47,30 @@ public void onFailure(@NonNull Call<RandomRecipesResponse> call, @NonNull Throwa
         });
     }
 
-    public void getSearchRecipes(RecipeDetailsListener listener, List<String> id){
+    public void getSearchRecipes(SearchRecipesListener listener, String query){
         SearchRecipesCall searchRecipesCall = retrofit.create(SearchRecipesCall.class);
-        Call<SearchRecipesResponse> call = searchRecipesCall.callSearchRecipes(id, "a0b47258ef634097812d0213ca6217ea");
+        Call<SearchRecipesResponse> call = searchRecipesCall.callSearchRecipes(query, null, null,null, null,"a0b47258ef634097812d0213ca6217ea");
```
Comment (JamesMcGaa) [inline app/src/main/java/com/example/metachef/RequestManager.java:52]: Hide your API keys, usually you define this in a file thats hidden by your .gitignore
https://stackoverflow.com/questions/57355856/how-to-hide-api-url-and-api-key-from-apiresources-java

Comment (ikechiafrics) [reply app/src/main/java/com/example/metachef/RequestManager.java:52]: Made these changes in (df2b500) commit

**Pull request merged**